CARDS-1774: Make the patient landing page text configurable

veronikaslc opened the pull request:
Added configuration page to the Admin dashboard

Commit:
CARDS-1774: Make the patient landing page text configurable
Added configuration page tot he Admin dashboard

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/WelcomeMessageConfiguration.jsx:
```diff
@@ -0,0 +1,137 @@
+    CardHeader,
+    List,
+    ListItem,
+	Typography
```
Wrong indentation

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/WelcomeMessageConfiguration.jsx:
```diff
@@ -0,0 +1,137 @@
+        setPath(json["@path"]);
+        setWelcomeMessage(json.text || "");
+      })
+	  .catch(setError);
```
Wrong indentation

marta- commented on modules/data-entry/src/main/frontend/src/questionnaireEditor/MarkdownText.jsx:
```diff
@@ -47,13 +47,14 @@ let MarkdownText = (props) => {
 
   return (
-    <MDEditor className={classes.markdown} value={value} onChange={value => {setValue(value); onChange && onChange(value);}} extraCommands={cmd}/>
+    <MDEditor className={classes.markdown} value={value} height={props.height || 200} onChange={value => {setValue(value); onChange && onChange(value);}} extraCommands={cmd}/>
```
Unpack `height` on line 43 and use `defaultProps` to specify the default if one is needed. Can we leave the default undefined? won't that keep the current look for all other occurrences of MDEditor?

marta- commented:
One usability concern is that what you see in the preview is not exactly what you get on the Patient Identification screen because of the `APP_NAME` placeholder. Let's do this:
* Add an informative text before the markdown editor explaining that APP_NAME will be replaced automatically with the application name: `In the welcome message, use APP_NAME to refer to the name configured for the application. On the Patient identification screen, all occurrences of APP_NAME will appear as {appName}`.
* Only show the editing side of the markdown editor
* On the right side of it render a `<FormattedText>` with the contents of the editor where we've replaced `APP_NAME` with the configured application name. Ideally this should be updated as you type.

Commit:
CARDS-1774: Make the patient landing page text configurable
Addressed codereview comments

marta- commented:
Looks great, with two small ui comments:
* the editor + preview together are narrower than the available width so there's an unnecessary gap on the right
* on narrow screens they remain side by side making the editor too narrow to use comfortably

Can we make the layout responsive, put them in a Grid and have the preview slide under the editor under 600px?

Commit:
CARDS-1774: Make the patient landing page text configurable
Made design more responsive

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/WelcomeMessageConfiguration.jsx:
```diff
@@ -0,0 +1,177 @@
+                  <Grid item xs={12} sm={8} md={6}>
+                    <MarkdownText value={welcomeMessage} height={350} preview="edit" onChange={value => { event.preventDefault(); setWelcomeMessage(value); }} />
+                  </Grid>
+                  <Grid item xs={12} sm={8} md={6}>
```
why `sm={8}` ? It just leaves unnecessary space on the sides.

marta- replied:
Removed in 46f7318.

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/WelcomeMessageConfiguration.jsx:
```diff
@@ -0,0 +1,177 @@
+                <Grid
+                  container
+                  spacing={2}
+                  direction={breakpoint ? "column" : "row"}
```
This breakpoint check is unnecessary, grids can wrap based on width.

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/WelcomeMessageConfiguration.jsx:
```diff
@@ -0,0 +1,177 @@
+  return (
+    <Card>
+      <CardHeader
+        title="Welcome Message text"
```
Let's be more specific: 
```suggestion
        title="Patient Portal Welcome Message"
```

marta- replied:
Updated in 204f653.

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/WelcomeMessageConfiguration.jsx:
```diff
@@ -0,0 +1,177 @@
+      <CardContent>
+        {error && <Typography color='error'>{error}</Typography>}
+        <Typography>
+          In the welcome message, use APP_NAME to refer to the name configured for the application.
```
Simpler and smaller description:
```suggestion
        <Typography variant="body2">
          Use APP_NAME to refer to the name configured for the application.
```

marta- replied:
Done in 204f653.

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/WelcomeMessageConfiguration.jsx:
```diff
@@ -0,0 +1,177 @@
+                        titleTypographyProps={{variant: "overline"}}
+                      />
+                      <CardContent>
+                        <FormattedText>
```
The preview font appears too large compared to the editor font, let's use a smaller one.
```suggestion
                        <FormattedText variant="body2">
```

marta- replied:
Done in bb8fe07.

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/WelcomeMessageConfiguration.jsx:
```diff
@@ -0,0 +1,177 @@
+                  spacing={2}
+                  direction={breakpoint ? "column" : "row"}
+                  justifyContent="center"
+                  alignItems="flex-start"
```
We should try to make the two boxes the same height for a more balanced look.
```suggestion
                  alignItems="stretch"
```

marta- replied:
Height match implemented in 470b0a2.

marta- commented on proms-resources/frontend/src/main/resources/SLING-INF/content/Extensions/Views/WelcomeMessageConfiguration.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "jcr:primaryType": "cards:Extension",
+  "cards:extensionPointId": "cards/coreUI/view",
+  "cards:extensionName": "Welcome Message Configuration",
```
```suggestion
  "cards:extensionName": "Patient Portal Welcome Message",
```

marta- replied:
Updated in 204f653.

marta- commented on proms-resources/frontend/src/main/resources/SLING-INF/content/Extensions/AdminDashboard/WelcomeMessageConfiguration.json:
```diff
@@ -0,0 +1,9 @@
+{
+  "jcr:primaryType": "cards:Extension",
+  "cards:extensionPointId": "cards/coreUI/admindashboard/entry",
+  "cards:extensionName": "Welcome Message Configuration Widget",
```
```suggestion
  "cards:extensionName": "Patient Portal Welcome Message",
```

marta- replied:
Updated in 204f653.

marta- commented on proms-resources/frontend/src/main/resources/SLING-INF/content/Extensions/AdminDashboard/WelcomeMessageConfiguration.json:
```diff
@@ -0,0 +1,9 @@
+  "cards:extensionName": "Welcome Message Configuration Widget",
+  "cards:targetURL": "/content.html/admin/WelcomeMessageConfiguration",
+  "cards:icon": "asset:proms-homepage.WelcomeMessageConfigurationIcon.js",
+  "cards:hint": "Configure the welcome message text",
```
```suggestion
  "cards:hint": "Configure the message displayed to Patient users on their login screen",
```

marta- replied:
Updated in 204f653.

marta- committed:
CARDS-1774: Make the patient landing page text configurable
Improve the configuration titles and hints

marta- committed:
CARDS-1774: Make the patient landing page text configurable
Decrease the preview font size to look more like the editor text.

marta- committed:
CARDS-1774: Make the patient landing page text configurable
Simplified Grid layout, removed unnecessary breakpoint logic

marta- committed:
CARDS-1774: Make the patient landing page text configurable
Made the editor and preview match height for a more balanced look
Hid unnecessary editor tools

sashaandjic reviewed (approved):
tested

veronikaslc committed:
CARDS-1774: Make the patient landing page text configurable

Removed unused path prop

sdumitriu reviewed (commented):
Bug, but not related to this PR: if you delete all the text and save, the patient login screen crashes.

Could be improved: after saving once, the button stays `SAVED` even if the text is changed again. I think it should go back to SAVE when the text changes.

sdumitriu committed:
CARDS-1774: Make the patient landing page text configurable
Bugfix: an empty message crashes the patient authentication page

sdumitriu committed:
CARDS-1774: Make the patient landing page text configurable
When the text is changed again after saving, set the button back to "Save" instead of "Saved"

sashaandjic reviewed (approved):
re-tested
Now if welcome text is blank, attempting to login as patient will not cause crashing, and the login dialog appear as expected.
![image](https://user-images.githubusercontent.com/23243244/175758911-491dc7ef-0951-4eee-942c-0f5585812e2b.png)
